### PR TITLE
Support specifying azure endpoints 

### PR DESCRIPTION
**Pull request by nalinigans**
Allow for AZURE_BLOB_ENDPOINT env to specify endpoints. Note that only https endpoints are supported for HIPPA/PHI compliance.
Also, start using azurite for testing with GitHub Actions.

### CODE REVIEW

File: core/src/storage_manager/storage_azure_blob.cc [hunk 1]
```diff
@@ -58,37 +58,71 @@
...
 
+static std::string run_command(const std::string& command) {
+  std::string output;
+  std::array<char, 2048> buffer;
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(command.data(), "r"), pclose);
+  if (pipe) {
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+      output += buffer.data();
+    }
+  }
```
Comment (kgururaj) [inline core/src/storage_manager/storage_azure_blob.cc:74]: Should there be an error if the if statement fails?

File: core/src/storage_manager/storage_azure_blob.cc [hunk 2]
```diff
@@ -58,37 +58,71 @@
...
+  // Try environment variables AZURE_STORAGE_ACCOUNT and AZURE_STORAGE_KEY first
   char *az_storage_ac_name = getenv("AZURE_STORAGE_ACCOUNT");
   if (az_storage_ac_name && account_name.compare(az_storage_ac_name) == 0) {
-    key = getenv("AZURE_STORAGE_KEY");
-  }  
-  return key;
+    char *key = getenv("AZURE_STORAGE_KEY");
+    if (key) {
+      return key;
+    }
```
Comment (kgururaj) [inline core/src/storage_manager/storage_azure_blob.cc:85]: Same comment

**Review by kgururaj — APPROVED** [review]

File: core/src/storage_manager/storage_azure_blob.cc (hunk 2 above)
Comment (nalinigans) [reply core/src/storage_manager/storage_azure_blob.cc:85]: If key is not found as env, we try again with the `az command` to get the key for the account.

File: core/src/storage_manager/storage_azure_blob.cc (hunk 1 above)
Comment (nalinigans) [reply core/src/storage_manager/storage_azure_blob.cc:74]: If the statement fails, output is empty and it is left to the callee to handle this. Would you like it handled differently?

**Review by mlathara — APPROVED** [review]
looks good

**Pull request merged**